hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -168,4 +175,29 @@ public String retryQuestion(String questionId, String comments, Question questio
                 .build());
         return askQuestion(question);
     }
+
+    public BinaryResource getAvatar(String agentId)
+    {
+        try
+        {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(format(config.getAvatarUrl(), agentId)))
+                    .headers(authService.getAuthHeaders())
+                    .GET()
+                    .build();
+
+            HttpResponse<InputStream> httpResponse = client.send(request, BodyHandlers.ofInputStream());
+
+            ensureCorrectHttpStatusReturned(SC_OK, httpResponse);
+            log.atDebug().log("Successfully retrieved avatar for Agent with id: {}", agentId);
+
+            String filePrefix = format("avatar-%s-%s", agentId, System.currentTimeMillis());
```

Comment:
Let's use a UUID to add randomness instead. It'd be safer than `currentTimeMillis`.

Reply:
Thank you, changed.